modules/t/Utils/FlatFile.t

```diff
@@ -24,15 +24,77 @@ use Test::Exception;
 use Bio::EnsEMBL::Compara::Utils::FlatFile;
 
 subtest 'map_row_to_header' => sub {
-    my $line = "a\tb\tc\td";
+    my $line_1 = "a\tb\tc\td";
+    my $line_2 = "a;b;c;d";
     my @header1 = (1, 2, 3, 4);
     my @header2 = (1, 2);
 
     is_deeply( 
-        Bio::EnsEMBL::Compara::Utils::FlatFile::map_row_to_header($line, \@header1), 
+        Bio::EnsEMBL::Compara::Utils::FlatFile::map_row_to_header($line_1, \@header1), 
         { 1 => 'a', 2 => 'b', 3 => 'c', 4 => 'd' }
     );
-    throws_ok { Bio::EnsEMBL::Compara::Utils::FlatFile::map_row_to_header($line, \@header2) } qr/Number of columns in header do not match row/, "Header doesn't match";
+
+    is_deeply( 
+        Bio::EnsEMBL::Compara::Utils::FlatFile::map_row_to_header($line_1, "1\t2\t3\t4"), 
+        { 1 => 'a', 2 => 'b', 3 => 'c', 4 => 'd' }
+    );
+    
+    is_deeply( 
+        Bio::EnsEMBL::Compara::Utils::FlatFile::map_row_to_header($line_1, \@header1, "\t"), 
+        { 1 => 'a', 2 => 'b', 3 => 'c', 4 => 'd' }
+    );
+
+    is_deeply( 
+        Bio::EnsEMBL::Compara::Utils::FlatFile::map_row_to_header($line_1, \@header1, '\t'), 
+        { 1 => 'a', 2 => 'b', 3 => 'c', 4 => 'd' }
+    );
```

Comment:
Both here and further down, I'd suggest to reconsider the repetition of tests for both single-quoted tab (`'\t'`) and double-quoted tab (`"\t"`). On the basis of the [Perl regex help page section on escape sequences](https://perldoc.perl.org/perlre#Escape-sequences) and how those escape sequences are handled in a regex, I'd argue that we can assume that if it works for one of these delimiter pattern strings, it will work for the other.